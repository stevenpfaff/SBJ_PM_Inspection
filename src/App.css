#root {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

#logo {
  height: 100px;
  width: 100px;
}

h1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

h2 {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-top: 1%;
}

table:first-of-type {
  margin-top: 1%;
}

table {
  margin: 3% 20%;
  width: 60% !important;
}

thead {
  text-align: center;
  background-color: #3565be;
  color: white;
}

tr > td:not(:first-of-type) {
  text-align: center;
}

.fixed-width {
  font-family: var(--bs-font-monospace);
}

td.distance {
  text-align: end !important;
}

td.duration {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}
