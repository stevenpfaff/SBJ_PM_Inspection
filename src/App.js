import './App.css'
import tasks from './Data/tasks.json'
import tasksii from './Data/tasksii.json'
import tasksiii from './Data/tasksiii.json'
import { Table } from 'react-bootstrap'
import logo from './sbj.png'

function App() {
  return (
    <div className="App">
      <h1 style={{ textAlign: 'center', marginTop: '3%' }}><img src={logo} height="100" width="100"></img>PM Inspection Checklist</h1>
      <h2 style={{ marginTop: '1%', marginLeft: '31%' }}>Work Order #</h2>
      <div style={{ marginTop: '1%', marginLeft: '20%', marginRight: '20%' }}>
        <Table striped bordered hover>
          <thead>
            <th>Customer</th>
            <th>Unit #</th>
            <th>VIN</th>
            <th>Miles</th>
            <th>Hours</th>
          </thead>
          <tbody>
            <tr>
              <td></td>
              <td></td>
              <td></td>
              <td></td>
              <td></td>
            </tr>
          </tbody>
        </Table>
      </div>
      <div style={{ marginTop: '3%', marginLeft: '20%', marginRight: '20%' }}>
        <Table striped bordered hover>
          <thead>
            <th>Task</th>
            <th>Completed</th>
            <th>Not Performed</th>
            <th>Part #</th>
          </thead>
          {tasks.map((task) => (
            <tbody>
              <tr>
                <td>{task.name}</td>
                <td><div>
                  <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
                </div></td>
                <td><div>
                  <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
                </div></td>
                <td></td>
              </tr>
            </tbody>
          ))}
        </Table>
      </div>
      <div style={{ marginTop: '3%', marginLeft: '20%', marginRight: '20%' }} >
        <Table striped bordered hover>
          <thead>
            <th>Task</th>
            <th>OK</th>
            <th>Adjusted</th>
            <th>Replace</th>
            <th>Monitor</th>
            <th>Not Performed</th>
          </thead>
          {tasksii.map((tasks) => (
            <tbody>
              <td>{tasks.name}</td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
            </tbody>
          ))}
        </Table>
      </div>
      <div style={{ marginTop: '3%', marginLeft: '20%', marginRight: '20%' }} >
        <Table striped bordered hover>
          <thead>
            <th>Task</th>
            <th>OK</th>
            <th>Location</th>
          </thead>
          {tasksiii.map((tasksiii) => (
            <tbody>
              <td>{tasksiii.name}</td>
              <td><div>
                <input class="form-check-input" type="checkbox" id="checkboxNoLabel" value="" aria-label="..." />
              </div></td>
              <td></td>
            </tbody>
          ))}
        </Table>
      </div>
    </div >
  );
}

export default App;
