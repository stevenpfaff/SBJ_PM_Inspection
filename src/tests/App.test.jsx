import { render } from '@testing-library/react'

import App from '../App'

describe('App', () => {
  it('should match snapshot', () => {
    const { container } = render(<App />)

    expect(container).toMatchSnapshot()
  })
})
