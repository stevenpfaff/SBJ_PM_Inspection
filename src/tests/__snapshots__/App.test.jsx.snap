// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <h1>
    <img
      id="logo"
      src="sbj.png"
    />
    PM Inspection Checklist
  </h1>
  <h2>
    Work Order #
    1234
  </h2>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Customer
        </th>
        <th>
          Unit #
        </th>
        <th>
          VIN
        </th>
        <th>
          Distance
        </th>
        <th>
          Duration
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jared Walters
        </td>
        <td
          class="fixed-width"
        >
          RP4ZJ0WPXX
        </td>
        <td
          class="fixed-width"
        >
          1G4GJ11Y9HP422546
        </td>
        <td
          class="distance"
        >
          300 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            13 mins
          </span>
        </td>
      </tr>
      <tr>
        <td>
          Stewart Garnett
        </td>
        <td
          class="fixed-width"
        >
          EFUK4D1K4M
        </td>
        <td
          class="fixed-width"
        >
          2GCEC19Z1S1244154
        </td>
        <td
          class="distance"
        >
          45 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            38 mins
          </span>
        </td>
      </tr>
      <tr>
        <td>
          Bruce Harvey
        </td>
        <td
          class="fixed-width"
        >
          YGM7GUJ7BJ
        </td>
        <td
          class="fixed-width"
        >
          WMWZN3C51BT133317
        </td>
        <td
          class="distance"
        >
          125 mi
        </td>
        <td
          class="duration"
        >
          <span>
            25 hrs
          </span>
          <span>
            20 mins
          </span>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Task
        </th>
        <th>
          Completed
        </th>
        <th>
          Not Performed
        </th>
        <th>
          Part #
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Oil Change
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          A320HGOEJ4
        </td>
      </tr>
      <tr>
        <td>
          Fuel Filters
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          6OF8WBFY6K
        </td>
      </tr>
      <tr>
        <td>
          Grease
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          M7R0S8PTRR
        </td>
      </tr>
      <tr>
        <td>
          Air Filter
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          HRXILKADTP
        </td>
      </tr>
      <tr>
        <td>
          Trans Service / Filter
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          GA9UPACDT9
        </td>
      </tr>
      <tr>
        <td>
          Axle Service
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          NUAI6HW36C
        </td>
      </tr>
      <tr>
        <td>
          Hydraulic Service / Filter
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          NO319JFOHF
        </td>
      </tr>
      <tr>
        <td>
          Drain Air Tanks
        </td>
        <td>
          <input
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          W221SNJFVX
        </td>
      </tr>
      <tr>
        <td>
          Apply PM Sticker
        </td>
        <td>
          <input
            checked=""
            data-testid="table-2-completed"
            type="checkbox"
          />
        </td>
        <td>
          <input
            data-testid="table-2-not-performed"
            type="checkbox"
          />
        </td>
        <td
          class="fixed-width"
        >
          67V86C503B
        </td>
      </tr>
    </tbody>
  </table>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Task
        </th>
        <th>
          OK
        </th>
        <th>
          Adjusted
        </th>
        <th>
          Replace
        </th>
        <th>
          Monitor
        </th>
        <th>
          Not Performed
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Tire Pressure / Condition
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Brakes
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Drive Shaft
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Springs
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Steering Components
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Coolant
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Power Steering
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check Lights
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check Strobe
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check Backup Alarm
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check Wipers
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check DOT
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check ANSI
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr>
        <td>
          Check FRA
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <div>
            <input
              type="checkbox"
            />
          </div>
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Task
        </th>
        <th>
          OK
        </th>
        <th>
          Location
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Engine Leaks
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Air Leaks
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Transmission Leaks
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Hydraulic Leaks
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Axle Leaks
        </td>
        <td>
          <input
            type="checkbox"
          />
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
